src/backend/distributed/utils/statistics_collection.c

```diff
@@ -86,7 +97,129 @@ CollectBasicUsageStatistics(void)
 	appendStringInfoString(fields, "}");
 
 	return SendHttpPostJsonRequest(STATS_COLLECTION_HOST "/v1/usage_reports",
-								   fields->data, HTTP_TIMEOUT_SECONDS);
+								   fields->data, HTTP_TIMEOUT_SECONDS, NULL);
+}
+
+
+/* CheckForUpdates queries Citus servers for newer releases of Citus. */
+void
+CheckForUpdates(void)
+{
+	SendHttpPostJsonRequest(
+		STATS_COLLECTION_HOST "/v1/releases/latest?flavor=community",
+		NULL, HTTP_TIMEOUT_SECONDS, &CheckForUpdatesCallback);
+}
+
+
+/*
+ * CheckForUpdatesCallback receives the response for the request sent by
+ * CheckForUpdates(). It processes the response, and if there is a newer release
+ * of Citus available, logs a LOG message. This function returns 0 if there are
+ * any errors in the received response, which means we didn't consume the data.
+ * Otherwise, it returns (size * count) which means we consumed all of the data.
+ */
+static size_t
+CheckForUpdatesCallback(char *contents, size_t size, size_t count, void *userData)
+{
+	PG_TRY();
+	{
+		const int citusVersionMajor = CITUS_VERSION_NUM / 10000;
+		const int citusVersionMinor = (CITUS_VERSION_NUM / 100) % 100;
+		const int citusVersionPatch = CITUS_VERSION_NUM % 100;
+		Datum responseJson = 0;
+		StringInfo releaseVersion = NULL;
+		int releaseMajor = 0;
+		int releaseMinor = 0;
+		int releasePatch = 0;
+		char *updateType = NULL;
+
+		StringInfo responseNullTerminated = makeStringInfo();
+		appendBinaryStringInfo(responseNullTerminated, contents, size * count);
+		responseJson = DirectFunctionCall1(jsonb_in,
+										   CStringGetDatum(responseNullTerminated->data));
+
+		if (!JsonbFieldExists(responseJson, "version") ||
+			!JsonbFieldExists(responseJson, "major") ||
+			!JsonbFieldExists(responseJson, "minor") ||
+			!JsonbFieldExists(responseJson, "patch"))
```

Comment:
These are to make sure ```jsonb_object_field_text``` doesn't return ```NULL```. We cannot use ```DirectFunctionCall``` for null-returning functions.

Reply:
```jsonb_object_field_text``` also works for object or array fields, but it returns null for ```null``` json values. If handling ```null``` values aren't easy in C code, should we skip handling them for now and leave it as TODO unless they cause crashes/etc.?